src/utils/config.ts

```diff
@@ -439,34 +439,108 @@ export function configAutoResolve(
  */
 export function configGetAllEnabled<T extends ConfigMetricAlarms = ConfigMetricAlarms>(
   confType: ConfigLocalType,
-  metrics: string[],
+  metrics?: string[],
 ): ConfigLocals<T> {
   const all = configGetAll(confType);
 
-  return Object.keys(all).reduce((acc, key) => {
-    const locals = all[key];
-
-    if (!locals) {
+  return Object.entries(all).reduce((acc, [key, allLocals]) => {
+    if (!allLocals) {
       return acc;
     }
 
-    const isEnabled = Object.keys(locals).find(key => {
-      if (!metrics.includes(key)) {
-        return false;
+    const locals = Object.entries(allLocals).reduce((acc, [key, local]) => {
+      if (metrics && !metrics.includes(key)) {
+        return acc;
       }
-      const local = locals[key];
-      if (local.enabled === false) {
-        return Object.values(local.alarm || {}).find(l => l.enabled === true) !== undefined;
+
+      if (!configIsEnabled(local)) {
+        return acc;
       }
-      return Object.values(local.alarm || {}).find(l => l.enabled !== false) !== undefined;
-    });
 
-    // Add only if at least one value is enabled
-    if (isEnabled) {
-      return { ...acc, [key]: locals };
-    }
+      return {
+        ...acc,
+        [key]: {
+          ...local,
+          alarms: Object.entries(local.alarm || {}).reduce((acc, [topic, alarm]) => {
+            if ((local.enabled === false && alarm.enabled === true) || alarm.enabled !== false) {
+              return {
+                ...acc,
+                [topic]: alarm,
+              };
+            }
+
+            return acc;
+          }, {} as TopicMap<AlarmOptions>),
+        }
+      }
+    }, {} as T);
 
     // Skip adding as none was enabled
-    return acc;
+    if (Object.keys(locals).length === 0) {
+      return acc;
+    }
+
+    return { ...acc, [key]: locals };
   }, {});
 }
+
+/**
+ * Calculate how many resources each metric alarm (example lambda) takes
+ */
+export function calculateResources<T extends ConfigMetricAlarms = ConfigMetricAlarms>(
+  confType: ConfigLocalType,
+  allowedMetrics?: string[],
+  addPerResource = 0,
+): { [key: string]: [number, T] } {
+  const enabled = configGetAllEnabled<T>(confType, allowedMetrics);
+  const metricCount = 1;
+
+  const acc: { [key: string]: [number, T] } = {};
+  for (const [key, metrics] of Object.entries(enabled)) {
+    for (const [metric, local] of Object.entries(metrics)) {
+      const [currentCount, currentLocals] = acc[key] || [0, {}]
+
+      const alarmCount = Object.values(local.alarm || {}).length;
+      acc[key] = [currentCount + alarmCount + metricCount + addPerResource, {
+        ...currentLocals,
+        [metric]: local
+      }];
+    }
+  }
+
+  return acc;
+}
+
+/**
+ * Generate chunked array of the metric alarms
+ */
+export function chunkByStackLimit<T extends ConfigMetricAlarms = ConfigMetricAlarms>(
```

Comment:
It would be a good idea to use verb in the naming here to make it more meaningful.

Reply:
Naming has now been improved to `resourceCountMap` and the function was also changed to `getResourceCountMap`